resolver/src/main/java/io/netty/resolver/DefaultHostsFileEntriesResolver.java

```diff
@@ -90,6 +89,17 @@ public InetAddress address(String inetHost, ResolvedAddressTypes resolvedAddress
         }
     }
 
+    private void ensureHostsFileEntriesAreFresh() {
+        long last = lastRefresh.get();
+        if (System.nanoTime() - last > refreshInterval) {
```

Comment:
nit: store `System.nanoTime()` in a local variable and reuse it in the compareAndSet. No need to call the function two times.

Reply:
fixed